internal/provider/resource/app_connection/app_connection_gitlab.go

```diff
@@ -0,0 +1,135 @@
+package resource
+
+import (
+	"context"
+	infisical "terraform-provider-infisical/internal/client"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+)
+
+type AppConnectionGitlabCredentialsModel struct {
+	AccessToken     types.String `tfsdk:"access_token"`
+	InstanceUrl     types.String `tfsdk:"instance_url"`
+	AccessTokenType types.String `tfsdk:"access_token_type"`
+}
+
+const AppConnectionGitlabAuthMethodAccessToken = "access-token"
+
+func NewAppConnectionGitlabResource() resource.Resource {
+	return &AppConnectionBaseResource{
+		App:               infisical.AppConnectionAppGitlab,
+		AppConnectionName: "GitLab",
+		ResourceTypeName:  "_app_connection_gitlab",
+		AllowedMethods:    []string{AppConnectionGitlabAuthMethodAccessToken},
+		CredentialsAttributes: map[string]schema.Attribute{
+			"access_token": schema.StringAttribute{
+				Required:    true,
+				Description: "The GitLab access token for authentication.",
+				Sensitive:   true,
+			},
+			"instance_url": schema.StringAttribute{
+				Required:    true,
+				Description: "The GitLab instance URL (e.g., https://gitlab.com for GitLab.com or your self-hosted GitLab URL).",
+			},
```

Comment:
this should be optional right? so if it's not explicitly passed in then it defaults to the GitLab url? 

<img width="749" height="817" alt="image" src="https://github.com/user-attachments/assets/cdc46180-7fe7-41e7-ae94-6e9017823463" />